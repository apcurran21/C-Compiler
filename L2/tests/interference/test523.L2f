(@array
	1
	%n <- rdi
	%first <- 1
	%index <- -1
	%check <- %n <= %index
	cjump %check = 1 :call_label_ciao_2
	%index += 1
	%first <<= 1
	%first += 1
	rdi <- %first
	call print 1
	%check <- %n = %index
	cjump %check = 1 :call_label_ciao_2
	%index += 1
	rdi <- %first
	call print 1
	%first <- 1
	%second <- 1
	:call_label_ciao_1
	%check <- %n = %index
	cjump %check = 1 :call_label_ciao_2
	%index += 1
	rdi <- %first
	rsi <- %second
	mem rsp -8 <- :call_label_ciao_3
	call @calculate 2
	:call_label_ciao_3
	%value <- rax
	%first <- %second
	%second <- %value
	%value <<= 1
	%value += 1
	rdi <- %value
	call print 1
	goto :call_label_ciao_1
	:call_label_ciao_2
	return
)
