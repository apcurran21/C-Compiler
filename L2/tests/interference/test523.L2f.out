%check %first %index %n r12 r13 r14 r15 rax rbp rbx %second
%first %check %index %n r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi %second %value
%index %check %first %n r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi %second %value
%n %check %first %index r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi %second %value
r10 %first %index %n r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi %second
r11 %first %index %n r10 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi %second
r12 %check %first %index %n r10 r11 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi %second %value
r13 %check %first %index %n r10 r11 r12 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi %second %value
r14 %check %first %index %n r10 r11 r12 r13 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi %second %value
r15 %check %first %index %n r10 r11 r12 r13 r14 r8 r9 rax rbp rbx rcx rdi rdx rsi %second %value
r8 %first %index %n r10 r11 r12 r13 r14 r15 r9 rax rbp rbx rcx rdi rdx rsi %second
r9 %first %index %n r10 r11 r12 r13 r14 r15 r8 rax rbp rbx rcx rdi rdx rsi %second
rax %check %first %index %n r10 r11 r12 r13 r14 r15 r8 r9 rbp rbx rcx rdi rdx rsi %second
rbp %check %first %index %n r10 r11 r12 r13 r14 r15 r8 r9 rax rbx rcx rdi rdx rsi %second %value
rbx %check %first %index %n r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rcx rdi rdx rsi %second %value
rcx %first %index %n r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rdi rdx rsi %second
rdi %first %index %n r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdx rsi %second
rdx %first %index %n r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rsi %second
rsi %first %index %n r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx %second
%second %check %first %index %n r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi %value
%value %first %index %n r12 r13 r14 r15 rbp rbx %second
